Source/DafnyLanguageServer/Workspace/NotificationPublisher.cs

```diff
@@ -1,16 +1,35 @@
-using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using Microsoft.Dafny.LanguageServer.Workspace.Notifications;
 using OmniSharp.Extensions.LanguageServer.Protocol.Document;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
 using OmniSharp.Extensions.LanguageServer.Protocol.Server;
 using System.Linq;
+using System.Net.Mime;
 using Microsoft.Extensions.Logging;
-using Serilog.Core;
 using Microsoft.Dafny.LanguageServer.Language;
+using Microsoft.Dafny.LanguageServer.Language.Symbols;
 using Microsoft.Extensions.Options;
+using OmniSharp.Extensions.LanguageServer.Protocol;
 
 namespace Microsoft.Dafny.LanguageServer.Workspace {
+  public record CompilationView(
```

Comment:
> Could you please move this record to its own file?

Good idea.

> Correct?

Yes. I'm not sure what the best name is! I chose 'View' because it's what is exposed externally to the client about the state of compilation. It could also be something that abstracts away from compilation, like `VisibleIdeState`